mongosql/src/ast/rewrites/mod.rs

```diff
@@ -86,6 +88,7 @@ pub fn rewrite_query(query: ast::Query) -> Result<ast::Query> {
         &OptionalParameterRewritePass,
         &NotComparisonRewritePass,
         &ScalarFunctionsRewritePass,
+        &WithQueryRewritePass,
```

Comment:
Note, we actually want to do WithQueryRewrite last because it can introduce duplicate queries, simple example:

```
WITH foo AS (...),
(SELECT * FROM foo JOIN foo)
```

If there are rewrites that apply in `...`, then they will be applied twice if we don't do WithQueryRewrite last.